main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala

```diff
@@ -986,6 +986,16 @@ object TypeInference {
           } yield (tconstrs, tpe, eff)
         }
 
+        // We special case the result type of the operation.
+        val operationType = operation.spec.tpe.typeConstructor match {
```

Comment:
What happens if we make the `Void` type as `fresh a` everywhere, instead of just here?

Reply:
Not sure exactly how we would do it, but it seems possible...

Reply:
Unsound

Reply:
What I have added is essentially sugar for `match do Foo.foo() {}`, i.e. case elimination on the empty type, but only for an effect operation.